pulser-core/pulser/sequence/sequence.py

```diff
@@ -205,21 +213,45 @@ def declared_variables(self) -> dict[str, Variable]:
     @property
     def available_channels(self) -> dict[str, Channel]:
         """Channels still available for declaration."""
-        # Show all channels if none are declared, otherwise filter depending
-        # on whether the sequence is working on XY mode
-        # If already in XY mode, filter right away
-        if not self._schedule and not self._in_xy:
-            return dict(self._device.channels)
+        all_channels = {**self._device.channels, **self._device.dmm_channels}
+        xy_channels = {
+            id: ch for id, ch in all_channels.items() if ch.basis == "XY"
+        }
+        if not self._schedule:
+            # Show all channels if none are declared
+            if not self._in_xy:
+                return all_channels
+            # If already in XY mode, filter right away but let the possibility
+            # to define an SLM Mask using a DMM channel
+            return {**xy_channels, **self._device.dmm_channels}
         else:
-            # MockDevice channels can be declared multiple times
             occupied_ch_ids = [cs.channel_id for cs in self._schedule.values()]
-            return {
+            # MockDevice channels can be declared multiple times
+            non_occupied_channels = {
                 id: ch
-                for id, ch in self._device.channels.items()
+                for id, ch in all_channels.items()
                 if (
                     id not in occupied_ch_ids or self._device.reusable_channels
                 )
-                and (ch.basis == "XY" if self._in_xy else ch.basis != "XY")
+            }
+            if occupied_ch_ids == [self._slm_mask_dmm]:
```

Comment:
In XY mode, don't we skip `config_detuning_map()` in `config_slm_mask()`, meaning the chosen `dmm` won't be in `occupied_ch_ids`?